org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/extensions/contentmodel/participants/XMLSchemaErrorCode.java

```diff
@@ -192,11 +194,27 @@ public static Range toLSPRange(XMLLocator location, XMLSchemaErrorCode code, Obj
 		case cvc_type_3_1_1:
 			return XMLPositionUtility.selectAllAttributes(offset, document);
 		case cvc_complex_type_2_1:
-		case cvc_type_3_1_3:
 		case cvc_elt_3_2_1:
 			return XMLPositionUtility.selectContent(offset, document);
+		case cvc_type_3_1_3:
+		case cvc_datatype_valid_1_2_1: {
+			
+			String attrValue = (String) arguments[0];

```

Comment:
Could you grab this value using the StringUtils helper: https://github.com/angelozerr/lsp4xml/blob/master/org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/utils/StringUtils.java#L355

It handles the case the object is not a String.